[Hub Generated] Review request for Microsoft.Subscription: Removed an unused API to make things simple, also corrected examples

sriramvu opened the pull request:
If you are a MSFT employee you can view your [work branch via this link](https://portal.azure-devex-tools.com/branch/sriramvu/azure-rest-api-specs/dev-subscription-Microsoft.Subscription-2018-11-01-preview...master/).

### Contribution checklist:
- [ ] I have reviewed the [documentation](https://github.com/Azure/azure-rest-api-specs#basics) for the workflow.
- [ ] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

sriramvu committed:
Removed unused API to make things simple, also corrected examples

AutorestCI commented:
# Automation for azure-sdk-for-python

AutorestCI commented:
# Automation for azure-sdk-for-go

sriramvu commented:
We have introduced the removed API as part of PR #7352 (around week back). 
There are 2 routes to create WebDirect / EA subscription - sticking with the one latest for simplicity.

azuresdkci commented:
Can one of the admins verify this patch?

navysingla commented on specification/subscription/resource-manager/Microsoft.Subscription/preview/2018-11-01-preview/examples/createCspSubscription.json:
```diff
@@ -11,7 +11,7 @@
     "202": {
       "headers": {
-        "Location": "/providers/Microsoft.Subscription/subscriptionOperations/aeb76128-ebde-4f5a-a9ed-e7bbf2dbc994?api-version=2018-11-01-preview",
+        "Location": "https://api-dogfood.resources.windows-int.net/providers/Microsoft.Subscription/subscriptionOperations/MzcwYzI3MDItNjA1MC00NTZkLTgyMTgtZjM0NDcyMjkwMTY0?api-version=2018-03-01-preview",
```
Use prod one

navysingla commented on specification/subscription/resource-manager/Microsoft.Subscription/preview/2018-11-01-preview/subscriptions.json:
```diff
@@ -60,7 +60,7 @@
+    "/providers/Microsoft.Billing/billingAccounts/{billingAccountName}/billingProfiles/{billingProfileName}/invoiceSections/{invoiceSectionName}/providers/Microsoft.Subscription/createSubscription": {
       "post": {
         "description": "The operation to create a new WebDirect or EA Azure subscription.",
         "operationId": "Subscriptions_CreateSubscription",
```
lets be consistent, use subscriptionfactory

navysingla commented on specification/subscription/resource-manager/Microsoft.Subscription/preview/2018-11-01-preview/subscriptions.json:
```diff
@@ -195,7 +128,7 @@
             "description": "The subscription creation parameters.",
             "required": true,
             "schema": {
-              "$ref": "#/definitions/ModernSubscriptionCreationParameters"
```
Don't change the existing parameter name, it might be breaking change for existing SDK users

sriramvu committed:
Taking comments to be consistent within this API version

sriramvu commented:
"public.rest-api-specs In progress — This check has started..." > running for last 5.5 hours.
Corresponding Azure pipeline details show completed (https://dev.azure.com/azure-sdk/public/_build/results?buildId=154250&view=results)

sriramvu committed:
Update createSubscription.json

Making minor change to rerun the checks as public.rest-api-specs check is reporting as running for more than a day when it has actually got completed.

sriramvu commented:
@xccc-msft can you please review and merge the PR.

sriramvu commented:
@ARM review team - can one of you please review the changes?
@xccc-msft - tried pinging ARM OnCall for this couple of days back. What is the approach we are supposed to take? It's been 11 days PR opened and 7 days on WaitForARMFeedback - please suggest.

xccc-msft commented:
@sriramvu I dropped message with this PR to ARM reviewers.

pilor reviewed (approved):
No actual API changes in this PR

sriramvu commented:
@xccc-msft - can you please merge the changes.
